coldfront/core/user/views_/request_hub_views.py

```diff
@@ -0,0 +1,395 @@
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.db.models import Q
+from django.views.generic.base import TemplateView
+
+from coldfront.core.allocation.models import (AllocationAttributeType,
+                                              AllocationUserAttribute,
+                                              AllocationRenewalRequest,
+                                              AllocationAdditionRequest)
+from coldfront.core.project.models import (ProjectUserRemovalRequest,
+                                           SavioProjectAllocationRequest,
+                                           VectorProjectAllocationRequest,
+                                           ProjectUserJoinRequest, Project)
+from coldfront.core.project.utils_.permissions_utils import \
+    is_user_manager_or_pi_of_project
+
+
+class RequestListItem:
+    """
+    Object to keep track of all variables used in for each request type
+    in the request hub
+    """
+
+    __slots__ = ['num', 'title', 'num_pending', 'table',
+                 'pending_queryset', 'complete_queryset',
+                 'button_path', 'button_text', 'id']
+
+    def __init__(self):
+        num = None
```

Comment:
I haven't used `__slots__` before: do we need these local variables as well?

Reply:
It didn't. I used __slots__ bc I wanted to restrict what fields could be added to the RequestListItem

Reply:
So can/should we remove them from `__init__`?

Reply:
I removed it